pixeltable/share/packager.py

```diff
@@ -69,138 +70,122 @@ def __init__(self, table: catalog.Table, additional_md: Optional[dict[str, Any]]
         _logger.info('Building archive.')
         bundle_path = self.__build_tarball()
         _logger.info(f'Packaging complete: {bundle_path}')
         return bundle_path
 
-    def __export_table(self, t: catalog.Table) -> None:
+    def __export_table(self, tv: catalog.TableVersion) -> None:
         """
-        Exports the data from `t` into an Iceberg table.
+        Exports the data from `t` into a Parquet table.
         """
-        # First generate a select list for the data we want to extract from `t`. This includes:
-        # - all stored columns, including computed columns;
-        # - errortype and errormsg fields whenever they're defined.
-        # We select only those columns that are defined in this table (columns inherited from ancestor tables will be
-        # handled separately).
-        # For media columns, we substitute `col.fileurl` so that we always get the URL (which may be a file:// URL;
-        # these will be specially handled later)
-        select_exprs: dict[str, exprs.Expr] = {}
-
-        # As we generate the select list, we construct a separate list of column types. We can't rely on df._schema
-        # to get the column types, since we'll be substituting `fileurl`s for media columns.
-        actual_col_types: list[ts.ColumnType] = []
-
-        for col_name, col in t._tbl_version.get().cols_by_name.items():
-            if not col.is_stored:
-                continue
-            if col.col_type.is_media_type():
-                select_exprs[col_name] = t[col_name].fileurl
-            else:
-                select_exprs[col_name] = t[col_name]
-            actual_col_types.append(col.col_type)
-            if col.records_errors:
-                select_exprs[f'{col_name}_errortype'] = t[col_name].errortype
-                actual_col_types.append(ts.StringType())
-                select_exprs[f'{col_name}_errormsg'] = t[col_name].errormsg
-                actual_col_types.append(ts.StringType())
-
-        # Run the select() on `self.table`, not `t`, so that we export only those rows that are actually present in
-        # `self.table`.
-        df = self.table.select(**select_exprs)
-        namespace = self.__iceberg_namespace(t)
-        self.iceberg_catalog.create_namespace_if_not_exists(namespace)
-        iceberg_schema = self.__to_iceberg_schema(df._schema)
-        iceberg_tbl = self.iceberg_catalog.create_table(f'{namespace}.{t._name}', schema=iceberg_schema)
-
-        # Populate the Iceberg table with data.
-        # The data is first loaded from the DataFrame into a sequence of pyarrow tables, batched in order to avoid
-        # excessive memory usage. The pyarrow tables are then amalgamated into the (single) Iceberg table on disk.
-        for pa_table in self.__to_pa_tables(df, actual_col_types, iceberg_schema):
-            iceberg_tbl.append(pa_table)
+        sql_types = {col.name: col.type for col in tv.store_tbl.sa_tbl.columns}
```

Comment:
Assert somewhere that tv is in the the tableversionpath of self.table?

Reply:
Done.